CMakeLists.txt

```diff
@@ -226,6 +227,9 @@ set (CMAKE_MACOSX_RPATH ON)
 # silence a CMake 3.11 warning that the old behavior is deprecated.
 cmake_policy (SET CMP0042 NEW)
 
+include_directories ("${PROJECT_BINARY_DIR}/src")
+include_directories ("${PROJECT_BINARY_DIR}/src/common")
```

Comment:
I believe the cause of `*-cse` tasks failing is that they are only building libbson, not libmongoc, and do not have the include path to the generated common-config.h.

I would move this into `src/libmongoc/CMakeLists.txt` next to the `include_directories` directive for the source common headers: https://github.com/mongodb/mongo-c-driver/blob/1.16.2/src/libmongoc/CMakeLists.txt/#L465

And similar for src/libbson/CMakeLists.txt.

Reply:
You can test this locally by configuring with `-DENABLE_MONGOC=OFF`:
```
cmake -DENABLE_MONGOC=OFF ..
cmake --build . --target bson_shared
```

Reply:
Ah, I looked into this locally a bit. I think you stumbled onto an existing issue.

Add `add_subdirectory (src/common)` just below `add_subdirectory (src/libbson)` on L236. If only libbson is being configured without libmongoc, cmake is currently ignoring the common directory (and hence, not generating the common header).

You can verify that works locally by configuring and building just `bson_shared`.